### PR TITLE
Fix log [Conn] Connection closed by peer (user)

### PR DESCRIPTION
**Pull request by slook**
- closed_connection [Conn] log message was garbled log output spanning multiple lines making the [Conn] log unreadable.
_With this commit it now prints:_
`17:04:46 [Conn] Connection of type P to user Username closed by peer ('12.34.56.78', 12345)`

- connect_to_peer_direct [Conn] log message changed wording because it's too soon to know if successful init will be possible or not. Also added IP address because it's inexpensive and useful here. _With this commit it now prints:_
`17:06:16 [Conn] Attempting direct connection of type P to user Username ('12.34.56.78', 12345)`

### CODE REVIEW

**Comment by slook** [comment]
I am unsure if error=None is needed on Line 986 for catching the correct events, or if it can be safely removed?
`    def closed_connection(self, conn, addr, error=None):`

The error is not useful to enter into the log, because it is often contextually incorrect (None) by this point.

**Comment by mathiascode** [comment]
> I am unsure if error=None is needed on Line 986 for catching the correct events, or if it can be safely removed?
> ` def closed_connection(self, conn, addr, error=None):`
> 
> The error is not useful to enter into the log, because "None" is often contextually incorrect by this point.

You can remove it, but you also have to do it at line 1064. Perhaps add a new log string around this line, something like "connection attempt to peer x failed. Error:".

File: pynicotine/pynicotine.py [hunk 1]
```diff
@@ -992,8 +993,8 @@ def closed_connection(self, conn, addr, error=None):
 
             for i in self.peerconns:
                 if i.conn == conn:
-                    log.add_conn("Connection closed by peer: %(peer)s. Error: %(error)s",
```
Comment (mathiascode) [inline pynicotine/pynicotine.py:995]: This reminds me, we should probably remove "by peer". The connection isn't necessary closed by the other user/peer, sometimes we have to do it too.

**Comment by slook** [comment]
Not sure if we will have a username if we cannot connect. I am unable to trigger this error on line 1064, do you know what scenario would lead up to this?

**Comment by mathiascode** [comment]
I forgot how this code worked. https://github.com/nicotine-plus/nicotine-plus/commit/36f484d2066c0a4c6886b9436dd6469d8705165f removes the confusing call at the end of connect_error, so you don't have to add a new log message.

**Comment by slook** [comment]
Oh, I already did and committed. Now there's conflicts I don't know how to resolve.

**Comment by mathiascode** [comment]
Could you merge the latest changes from master to your branch, so that previous changes such as line 1062-1064 don't appear in the diff?

**Comment by slook** [comment]
Done. Sorry my branch is a mess, I don't know the cleanest way of going about things.

**Comment by mathiascode** [comment]
No problem.

**Review by mathiascode — APPROVED** [review]

**Pull request merged**